pair-finder/pair_finder.py

```diff
@@ -145,6 +147,11 @@ def _thread_main(repo, task_name, log_level, skip_if_output_exists, keep_clone):
     pipeline = Pipeline(steps)
 
     result, out_context = pipeline.run(None, in_context)
+    if not result:
+        # A filter in the pipeline encountered a fatal error and made the pipeline exit early.
+        # Skip writing the output file.
+        log.info('Exiting the program as there are no jobs to continue mining.')
+        sys.exit(1)
```

Comment:
Is the only case that `result` will be `None` when there are no jobs?

Reply:
Yes, if there are no jobs we will raise a `StepException` error which will return None.